### PR TITLE
Refactor: created a new module for uniquecategories

### PR DESCRIPTION
**Pull request by Tonie-NG**
Fixes issue #63

### CODE REVIEW

File: frontend/src/components/categories/dropdown.tsx [hunk 1]
```diff
@@ -110,7 +107,7 @@ export const Location = () => {
         background="var(--deep-charcoal)"
         border="1px solid var(--bg-two)"
       >
-        {locations?.map((location, index) => {
+        {locations?.map((location: string, index: any) => {
```
Comment (kaf-lamed-beyt) [inline frontend/src/components/categories/dropdown.tsx:110]: Let's allow `index` to receive a type of `number`

**Review by kaf-lamed-beyt — CHANGES_REQUESTED** [review]
This looks beautifully hand-crafted, my chief! 🤩 

Loved the way you modularized the functions even to the smallest occurrence — `addCategory()`

File: frontend/src/components/categories/dropdown.tsx (hunk 1 above)
Comment (Tonie-NG) [reply frontend/src/components/categories/dropdown.tsx:110]: Alright. I'm on it

Comment (kaf-lamed-beyt) [reply frontend/src/components/categories/dropdown.tsx:110]: Thanks chief!!🙌🏽

Comment (Tonie-NG) [reply frontend/src/components/categories/dropdown.tsx]: I've done that.

**Review by kaf-lamed-beyt — APPROVED** [review]
Bless up! @Tonie-NG 🚀

**Pull request merged**